packages/plugins/auto-doc/src/stores/doc-templates.store.ts

```diff
@@ -0,0 +1,216 @@
+import { v4 as uuidv4 } from 'uuid';
+// OPENSCD
+import { createElement } from '@oscd-plugins/core'
+// SVELTE
+import { writable, get } from 'svelte/store'
+// STORES
+import { pluginStore, eventStore } from './index'
+
+//====== STORES ======//
+const { xmlDocument } = pluginStore
+
+//==== STATE
+const autoDocArea = writable<Element | null>(null);
+
+//==== PRIVATE ACTIONS
+function setPrivateArea() {
+    const xmlDoc = get(xmlDocument);
+    if (!xmlDoc) {
+        throw new Error("XML Document is not defined");
+    }
+    const newPrivateArea = createPrivateArea(xmlDoc);
+    autoDocArea.set(newPrivateArea);
+
+}
+
+function createPrivateArea(xmlDocument: Document): Element {
+    const rootElement = xmlDocument.documentElement;
+    const existingPrivateArea = rootElement.querySelector('Private[type="AUTO_DOC"]');
+    if (existingPrivateArea) {
+        return existingPrivateArea;
+    }
+
+    const newPrivateArea = createElement(xmlDocument, 'Private', { type: 'AUTO_DOC' });
+    rootElement.appendChild(newPrivateArea);
+
+    eventStore.createAndDispatchActionEvent(rootElement, newPrivateArea);
+    return newPrivateArea;
+}
+
+function insertBlockAtPosition(docDef: Element, blockIdElement: Element, position: number) {
+    if (position >= 0 && position < docDef.children.length) {
+        docDef.insertBefore(blockIdElement, docDef.children[position]);
+    } else {
+        docDef.appendChild(blockIdElement);
+    }
+}
+
+//==== PUBLIC ACTIONS
+function getDocumentTemplate(id: string): Element | null {
+    const currentPrivateArea = get(autoDocArea);
+    if (currentPrivateArea) {
+        return currentPrivateArea.querySelector(`DocumentTemplate[id="${id}"]`);
+    }
+    return null;
+}
+
+function getAllDocumentTemplates() {
+    const currentPrivateArea = get(autoDocArea);
+    if (currentPrivateArea) {
+        return Array.from(currentPrivateArea.querySelectorAll('DocumentTemplate'));
+    }
+    return [];
+}
+
+function getAllBlocksOfDocumentTemplate(docTemplateId: string) {
+    const docTemplate = getDocumentTemplate(docTemplateId);
+    if (docTemplate) {
+        return Array.from(docTemplate.querySelectorAll('Block'));
+    }
+    return [];
+}
+
+function getBlockOfDocumentTemplate(docTemplateId: string, blockId: string) {
+    const docTemplate = getDocumentTemplate(docTemplateId);
+    if (docTemplate) {
+        return docTemplate.querySelector(`Block[id="${blockId}"]`);
+    }
+    return null;
+}
+
+function addDocumentTemplate(description: string): string | null {
+    let generatedId: string | null = null;
+    const xmlDoc = get(xmlDocument);
+    if (!xmlDoc) {
+        throw new Error("XML Document is not defined");
+    }
+
+    const currentPrivateArea = get(autoDocArea);
+    if (currentPrivateArea) {
+        const id = uuidv4();
+
+        const newDocDef = createElement(xmlDoc, 'DocumentTemplate', {
+            id: id,
+            date: new Date().toISOString(),
+            description: description
```

Comment:
Could it be that we might want to add a title as well?

Reply:
yes you are right. I'll add it